scipy/stats/_multivariate.py

```diff
@@ -5739,3 +5741,632 @@ def _sample_uniform_direction(dim, size, random_state):
+        return samples
+
+    def _rotate_samples(self, samples, mu, dim):
+        """A QR decomposition is used to find the rotation that maps the
+        north pole (1, 0,...,0) to the vector mu. This rotation is then
+        applied to all samples.
+
+        Parameters
+        ----------
+        samples: array_like, shape = [..., n]
+        mu : array-like, shape=[n, ]
+            Point to parametrise the rotation.
+
+        Returns
+        -------
+        rotation_matrix : array-like, shape=[n, n]
+            Rotation matrix to rotate vectors from (1, 0, ..., 0)
+            to mu
+        sign : 1 or -1
+            Sign the vectors have to be multiplied with after rotation
+        """
+        base_point = np.zeros((dim, ))
+        base_point[0] = 1.
+        embedded = np.concatenate([mu[None, :], np.zeros((dim - 1, dim))])
+        rotmatrix, _ = np.linalg.qr(np.transpose(embedded))
+        if np.allclose(np.matmul(rotmatrix, base_point[:, None])[:, 0], mu):
+            rotsign = 1
+        else:
+            rotsign = -1
+
+        # apply rotation
+        samples = np.einsum('ij,...j->...i', rotmatrix, samples) * rotsign
+        return samples
+
+    def _rvs(self, dim, mu, kappa, size, random_state):
+        if dim == 2:
+            samples = self._rvs_2d(mu, kappa, size, random_state)
+        else:
+            if dim == 3:
+                samples = self._rvs_3d(kappa, size, random_state)
+            else:
+                samples = self._rejection_sampling(dim, kappa, size,
+                                                   random_state)
+            samples = self._rotate_samples(samples, mu, dim)
+        return samples
+
+    def rvs(self, mu=None, kappa=1, size=1, random_state=None):
+        """Draw random samples from a von Mises-Fisher distribution.
+
+        Parameters
+        ----------
+        size : integer, optional
+            Number of samples to draw (default 1).
+
+        Returns
+        -------
+        rvs : ndarray or scalar
+            Random variates of size (`size`, `N`), where `N` is the
+            dimension of the random variable.
+
+        """
+        dim, mu, kappa = self._process_parameters(mu, kappa)
+        random_state = self._get_random_state(random_state)
+        samples = self._rvs(dim, mu, kappa, size, random_state)
+        return samples
+
+    def _entropy(self, dim, kappa):
+        halfdim = 0.5 * dim
+        return (-self._log_norm_factor(dim, kappa) - kappa *
+                ive(halfdim, kappa) / ive(halfdim - 1, kappa))
+
+    def entropy(self, mu=None, kappa=1):
+        """Compute the differential entropy of the von Mises-Fisher
+        distribution.
+
+        Parameters
+        ----------
+
+        Returns
+        -------
+        h : scalar
+            Entropy of the von Mises-Fisher distribution
+
+        """
+        dim, mu, kappa = self._process_parameters(mu, kappa)
+        return self._entropy(dim, kappa)
+
+    def fit(self, x):
```

Comment:
For univariate distributions, we offer the option of fixing parameters (e.g. user specifies location, SciPy fits scale). Is that a possibility here?

Reply:
Mathematically it is quite easy here: $\mu$ and $\kappa$ are estimated independently from each other. The question is if we want this also for multivariate distributions.

I could add the keywords `mu=None, kappa=None`. If one is set, it will not be fitted and the function returns the provided value like for univariate distributions.

Reply:
I prefer that to the `f0`, `f1`... for fixing shapes of univariate distributions.

Reply:
Puh, I am not fluent enough in the theory to potentially apply bias correction here. What about we stick with MLE and state in the docstring that no bias correction is applied?

Reply:
I'm OK with that since it's consistent with the univariate distributions, and it seems there are few distributions for which analytical bias-correction results are available.